src/hardware/disk_noise.cpp

```diff
@@ -0,0 +1,456 @@
+			std::ifstream file(path, std::ios::binary);
+			if (!file.is_open()) {
+				LOG_WARNING("DiskNoiseDevice: Failed to open %s",
+				            path.c_str());
+				return;
+			}
+
+			// Skip 44 bytes of the WAV header
+			char header[44];
+			file.read(header, 44);
+
+			std::vector<int16_t> temp;
+			int16_t sample;
+			while (file.read(reinterpret_cast<char*>(&sample),
+			                 sizeof(sample))) {
+				temp.push_back(sample);
+			}
+			LOG_DEBUG("DiskNoiseDevice: Loaded %zu samples from %s",
+			          temp.size(),
+			          path.c_str());
+			buffer = std::move(temp);
+			return;
+		}
+	}
+}
+
+void DiskNoiseDevice::LoadSeekSamples(const std::vector<std::string>& paths)
+{
+	for (const auto& path : paths) {
+		if (path.empty()) {
+			seek_samples_.emplace_back(); // Placeholder for missing
+			                              // sample
+			continue;
+		}
+
+		std::vector<int16_t> sample;
+		LoadSample(path, sample);
+		if (!sample.empty()) {
+			seek_samples_.push_back(std::move(sample));
+		} else {
+			seek_samples_.emplace_back();
+		}
+	}
+
+	// Assign weights with descending priority (most weight to first samples)
+	const size_t max_weight = 10;
+	seek_sample_weights_.resize(seek_samples_.size());
+	for (size_t i = 0; i < seek_samples_.size(); ++i) {
+		seek_sample_weights_[i] = static_cast<size_t>(max_weight - i);
+	}
+}
+
+size_t DiskNoiseDevice::ChooseWeightedSeekIndex() const
+{
+	size_t total_weight = 0;
+	for (size_t i = 0; i < seek_samples_.size(); ++i) {
+		if (!seek_samples_[i].empty()) {
+			total_weight += seek_sample_weights_[i];
+		}
+	}
+
+	if (total_weight == 0) {
+		return 0;
+	}
+
+	const size_t r = static_cast<size_t>(rand()) % total_weight;
+	size_t sum     = 0;
+	for (size_t i = 0; i < seek_samples_.size(); ++i) {
+		if (seek_samples_[i].empty()) {
+			continue;
+		}
+
+		sum += seek_sample_weights_[i];
+		if (r < sum) {
+			return i;
+		}
+	}
+	return 0;
+}
+
+DiskNoiseDevice::DiskNoiseDevice(const bool enable_disk_noise,
+                                 const std::string& spin_up_sample_path,
+                                 const std::string& spin_sample_path,
+                                 const std::vector<std::string>& seek_sample_paths,
+                                 const std::string& spin_channel_name,
+                                 const std::string& seek_channel_name,
+                                 const unsigned int& spin_volume,
+                                 const unsigned int& seek_volume)
+        : enable_disk_noise_(),
+          spin_up_sample_(),
+          spin_sample_(),
+          current_seek_sample_(),
+          seek_sample_weights_(),
+          seek_samples_(),
+          spin_pos_(0),
+          seek_pos_(0),
+          last_activity_(0),
+          spin_channel_(nullptr),
+          seek_channel_(nullptr)
+{
+	enable_disk_noise_ = enable_disk_noise;
+	if (!enable_disk_noise_) {
+		LOG_INFO("DiskNoiseDevice: Disk noise emulation disabled");
```

Comment:
Use the `DISKNOISE` prefix, please. Then we have consistency between the log prefix and the single `DISKNOISE` mixer channel.